### ISSUE TITLE
[apex] ApexCRUDViolation ignores Schema.isDeletable on Delete DML

### ISSUE THREAD

**Issue by rsoesemann** [pmd/pmd#4997]
**Affects PMD Version:** 
7.0.1 (and all versions before)

**Rule:**
ApexCRUDViolation - https://pmd.github.io/pmd/pmd_rules_apex_security.html#apexcrudviolation

**Description:**
PMD incorrectly reports the need for validation of CRUD permissions before a DML Delete statement although the Schema.isDeletable is called.

This issues was raised in https://salesforce.stackexchange.com/questions/421548/apexcrudviolation-check-for-object-level-delete-permission-does-not-work where the user used PMD inside the Salesforce Code Analyser CLI tool (uses PMD internally)

**Code Sample demonstrating the issue:**

```java
public static void exampleMethod() {
        List<Account> accounts = [SELECT Id FROM Account WITH SECURITY_ENFORCED];
        if (Schema.SObjectType.Account.isDeletable()) {
            delete accounts;
        }
    }
```

**Expected outcome:**

PMD should report no violation.

**Running PMD through:** Salesforce Code Analyser CLI https://developer.salesforce.com/docs/platform/salesforce-code-analyzer/overview using:

```
sf scanner run --format csv --outfile CodeAnalyzerGeneral.csv --target ./ --category Security
```

**Comment by rsoesemann** [comment]
Related Ticket on the Code Analyser repo is  https://github.com/forcedotcom/sfdx-scanner/issues/1458

**Comment by FishOfPrey** [comment]
In this particular scenario the best practice would be be `delete as user accounts;`

Any `WITH SECURITY_ENFORCED` scenarios have been superseded by the equivalent user mode operation.

There are scenarios where Security.[stripInaccessible](https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_class_System_Security.htm#topic-title) may make more sense, such as being able to selectively handle what occurs if the user doesn't have access. Although for a delete operation that would need to be a schema describe check.

**Comment by jh480** [comment]
Schema checks should not be discouraged, as ("managed") code within app-store components (esp. triggers) should not let users with unsufficient access run into errors (worst-case-scenario: users cannot edit their own User configurations).
